### ISSUE TITLE
sam build  does not work - probably library dependencies 

### ISSUE THREAD

**Issue by szymic1** [aws/aws-sam-cli#3792]
### Description:
After installation new aws-sam (1.43) we could not build lambdas, following error appears:

```
Building....
Traceback (most recent call last):
  File "/home/prod/trustedtwin-main/venv/bin/sam", line 8, in <module>
    sys.exit(cli())
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 166, in wrapped
    raise exception  # pylint: disable=raising-bad-type
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 124, in wrapped
    return_value = func(*args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/lib/utils/version_checker.py", line 41, in wrapped
    actual_result = func(*args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/cli/main.py", line 87, in wrapper
    return func(*args, **kwargs)
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/commands/build/command.py", line 174, in cli
    do_cli(
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/commands/build/command.py", line 220, in do_cli
    from samcli.commands.build.build_context import BuildContext
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/commands/build/build_context.py", line 13, in <module>
    from samcli.lib.providers.sam_api_provider import SamApiProvider
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/lib/providers/sam_api_provider.py", line 6, in <module>
    from samcli.lib.providers.api_collector import ApiCollector
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/lib/providers/api_collector.py", line 10, in <module>
    from samcli.local.apigw.local_apigw_service import Route
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/samcli/local/apigw/local_apigw_service.py", line 11, in <module>
    from flask import Flask, request
  File "/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/flask/__init__.py", line 14, in <module>
    from jinja2 import escape
ImportError: cannot import name 'escape' from 'jinja2' (/home/prod/trustedtwin-main/venv/lib/python3.9/site-packages/jinja2/__init__.py)
```

What helps to set requirements:
itsdangerous==2.0.1
werkzeug==2.0.2
Jinja2==3.0.3

**Comment by hawflau** [comment]
Hey @szymic1, how did you install SAM CLI and which OS are you on?

**Comment by szymic1** [comment]
I've install using pip, we installed on Ubuntu and Aws Linux.

**Comment by qingchm** [comment]
@szymic1 Thanks for reporting this in! This is due to the newest versions of jinja2 deprecated the "escape" as referenced in https://jinja.palletsprojects.com/en/3.1.x/changes/#version-3-1-0,  https://stackoverflow.com/questions/71718167/importerror-cannot-import-name-escape-from-jinja2 also has some context on this to read about, we will look into how to update our dependency versions to prevent this. Thanks for reporting this and we for sure will try to address this ASAP!

**Comment by qingchm** [comment]
@szymic1 Would you mind providing the template for us to reproduce in different environments?

**Comment by szymic1** [comment]
What do you mean about "templeta"? Our project where we use sam is quite complicated and it is hard to isolate 'sam' execution.

**Comment by qingchm** [comment]
@szymic1 I see, I would first recommend try installing SAM using brew or installing it in a separate virtual environment as a temporary workaround to unblock yourself. But the issue has been noted and meanwhile we will try to address it! Let me know if the workaround works for you!

**Comment by szymic1** [comment]
@qingchm so far we use this solution where we set:
itsdangerous==2.0.1
werkzeug==2.0.2
Jinja2==3.0.3
It works and we can wait for next SAM version :)

**Comment by qingchm** [comment]
@szymic1 Nice! I'll let you know when we have addressed this!

**Comment by qingchm** [comment]
I believe that this is fixed in the new 1.53.0 release. Please have a try and if you still have any questions or issues feel free to re-open the issue!
